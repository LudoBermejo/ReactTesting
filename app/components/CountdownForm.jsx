import React from 'react';

export default class CountdownForm extends React.Component {

  constructor(props) {
    super(props);
    this.onSubmitForm = this.onSubmitForm.bind(this);
  }

  onSubmitForm(e) {
    e.preventDefault();
    const strSeconds = this.textInput.value;

    if (strSeconds.match(/^[0-9]+$/)) {
      this.props.onStartCountdown(parseInt(strSeconds, 10));
      this.textInput.value = '';
    }
  }

  render() {
    return (
      <div>
        <form onSubmit={this.onSubmitForm}>
          <input
            type="text"
            placeholder="Enter type in seconds"
            ref={(input) => {
              this.textInput = input;
            }}
          />
          <button
            type="submit"
            className="button expanded"
          >Start</button>
        </form>
      </div>
    );
  }
}


CountdownForm.propTypes = {
  onStartCountdown: React.PropTypes.func.isRequired
};
