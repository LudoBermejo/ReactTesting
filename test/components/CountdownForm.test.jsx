import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import CountdownForm from 'CountdownForm';

describe('Countdown form component', () => {
  it('renders correctly', () => {
    function itRenders() {

    }
    const tree = renderer.create(<CountdownForm onStartCountdown={itRenders} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('not launch function if invalid value is sent', () => {
    const onButtonClick = jest.fn();

    const countdownForm = shallow(<CountdownForm onStartCountdown={onButtonClick} />);
    countdownForm.find('button').simulate('click');
    expect(onButtonClick).not.toHaveBeenCalled();
  });

  it('not launch function if valid value is sent', () => {
    const onButtonClick = jest.fn();

    const countdownForm = mount(<CountdownForm onStartCountdown={onButtonClick} />);
    console.log(countdownForm.ref('input').get(0));
    console.log(countdownForm.find('input').get(0).value);

    countdownForm.find('button').simulate('click');
    expect(onButtonClick).toHaveBeenCalled();
   });
});
