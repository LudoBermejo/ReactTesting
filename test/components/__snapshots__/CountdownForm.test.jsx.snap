exports[`Countdown form component renders correctly 1`] = `
<div>
  <form
    onSubmit={[Function]}>
    <input
      placeholder="Enter type in seconds"
      type="text" />
    <button
      className="button expanded"
      type="submit">
      Start
    </button>
  </form>
</div>
`;
